pkg/util/ed25519.go

```diff
@@ -0,0 +1,31 @@
+package util
+
+import (
+	"crypto/ed25519"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+)
+
+func ParseEd25519PrivateKey(pemString string) (ed25519.PrivateKey, error) {
+	if len(pemString) == 0 {
+		return nil, nil
+	}
+	block, _ := pem.Decode([]byte(pemString))
+	if block == nil {
+		return nil, errors.New("failed to parse PEM block containing the private key")
+	}
+
+	// Parse the private key
+	privateKeyInterface, err := x509.ParsePKCS8PrivateKey(block.Bytes)
```

Comment:
we seldom use the "Interface" in the variable name in golang, you can call it "pkcs8Key" here or just "privateKey"

Reply:
How about `anyPrivateKey`?